pkg/daemon/ceph/osd/kms/vault.go

```diff
@@ -129,12 +129,24 @@ func configTLS(clusterdContext *clusterd.Context, namespace string, config map[s
 }
 
 func put(v secrets.Secrets, secretName, secretValue string, keyContext map[string]string) error {
+	// First we must see if the key entry already exists, if it does we do nothing
+	key, err := get(v, secretName, keyContext)
+	if err != nil {
+		if err != secrets.ErrSecretExists || err != secrets.ErrInvalidSecretId {
```

Comment:
Did you mean `and`? Otherwise, it seems this will always return the error when `err != nil`
```suggestion
		if err != secrets.ErrSecretExists && err != secrets.ErrInvalidSecretId {
```

Reply:
Actually, this was the wrong approach, new version.